main.tf

```diff
@@ -1,136 +1,179 @@
+    [linux:children]
+    bootstraps
+    masters
+    agents_public
+    agents_private
+    [windows:children]
+    win_agents
+    EOF
+  }
 
-  root_block_device {
-    volume_size           = "${var.root_volume_size}"
-    volume_type           = "${var.root_volume_type}"
-    delete_on_termination = true
+  provisioner "file" {
+    source                           = "${path.module}/files/"
+    destination                      = "/tmp"
   }
 
-  user_data = "${var.user_data}"
 
-  lifecycle {
-    ignore_changes = ["user_data", "ami"]
+  provisioner "remote-exec" {
+    inline                         = [
+      "rm -rf /tmp/dcos-ansible ; sudo yum install git -y && cd /tmp && git clone -b feature/windows-agent-support https://github.com/alekspv/dcos-ansible && cd /tmp/dcos-ansible && sudo docker build -t dcos-ansible-bundle-win . && sudo docker run -it -v /tmp/win_inventory:/inventory -v /tmp/ansible.cfg:/ansible.cfg  dcos-ansible-bundle-win ansible-playbook -i inventory  -l win_agents dcos_playbook.yml"
+    ]
   }
-}
 
-resource "aws_ebs_volume" "volume" {
-  # We group volumes from one instance first. For instance:
-  # - length(var.extra_volumes) = 2 (volumes)
-  # - var.num = 3 (instances)
-  #
-  # We will get:
-  # - volume.0 (instance 0)
-  # - volume.1 (instance 0)
-  # - volume.2 (instance 1)
-  # - volume.3 (instance 1)
-  # - volume.4 (instance 2)
-  # - volume.5 (instance 2)
-  count = "${var.num * length(var.extra_volumes)}"
-
-  availability_zone = "${element(aws_instance.instance.*.availability_zone, count.index / local.num_extra_volumes)}"
-  size              = "${lookup(local.extra_volumes[count.index % local.num_extra_volumes], "size", "120")}"
-  type              = "${lookup(local.extra_volumes[count.index % local.num_extra_volumes], "type", "")}"
-  iops              = "${lookup(local.extra_volumes[count.index % local.num_extra_volumes], "iops", "0")}"
-
-  tags = "${merge(var.tags, map(
-                "Name", format(var.extra_volume_name_format,
-                               var.cluster_name,
-                               element(aws_instance.instance.*.id, count.index / local.num_extra_volumes)),
-                "Cluster", var.cluster_name))}"
 }
 
-resource "aws_volume_attachment" "volume-attachment" {
-  count        = "${var.num * length(var.extra_volumes)}"
-  device_name  = "${lookup(local.extra_volumes[count.index % local.num_extra_volumes], "device_name", "dummy")}"
-  volume_id    = "${element(aws_ebs_volume.volume.*.id, count.index)}"
-  instance_id  = "${element(aws_instance.instance.*.id, count.index / local.num_extra_volumes)}"
-  force_detach = true
+resource "null_resource" "pass" {
+  count                              = "${var.num_winagent}"
+  triggers                           = {
+var.adm = "${aws_instance.mesos-agent-windows.*.private_ip[count.index]}   pass=${rsadecrypt(aws_instance.mesos-agent-windows.*.password_data[count.index], file("${var.ssh_private_key_file}"))}"
```

Comment:
users might not want to state their key for this. Wouldn't a randomly generated key be a better solution for this? In general it has nothing to do with the way we authenticate against those agents.

Or if possible can we provide a generated password by cloud-init?

Even better would be a certificate based authentication...